casa_formats_io/casa_low_level_io.py

```diff
@@ -113,329 +216,889 @@ def read_type(f):
+        Record.read(f)
 
-    desc = {}
+        flag = f.read(1)  # noqa
 
-    desc['_keywords_'] = read_table_record(f, image_path)
-    desc['_define_hypercolumn_'] = {}
+        ndim2 = read_int32(f)  # noqa
 
-    hypercolumn = read_table_record(f, image_path)
-    desc['_private_keywords_'] = hypercolumn
-    if hypercolumn:
-        name = list(hypercolumn)[0].split('_')[1]
-        value = list(hypercolumn.values())[0]
-        desc['_define_hypercolumn_'][name] = {'HCcoordnames': value['coord'],
-                                              'HCdatanames': value['data'],
-                                              'HCidnames': value['id'],
-                                              'HCndim': value['ndim']}
+        self.cube_shape = read_iposition(f)
+        self.tile_shape = read_iposition(f)
 
-    ncol = read_int32(f)
+        unknown = read_int32(f)  # noqa
+        unknown = read_int32(f)  # noqa
 
-    for icol in range(ncol):
-        if icol > 0:
-            read_int32(f)
-        array_column_desc = read_column_desc(f, image_path)
-        desc.update(array_column_desc)
 
-    return desc
+class TiledCellStMan(TiledStMan):
 
+    @classmethod
+    def read(cls, f):
+        self = cls()
+        self.name = 'TiledCellStMan'
+        return self
 
-@with_nbytes_prefix
-def read_tiled_st_man(f):
+    @with_nbytes_prefix
+    def read_header(self, f):
 
-    # The code in this function corresponds to TiledStMan::headerFileGet
-    # https://github.com/casacore/casacore/blob/75b358be47039250e03e5042210cbc60beaaf6e4/tables/DataMan/TiledStMan.cc#L1086
+        # The code in this function corresponds to TiledStMan::headerFileGet
+        # https://github.com/casacore/casacore/blob/75b358be47039250e03e5042210cbc60beaaf6e4/tables/DataMan/TiledStMan.cc#L1086
 
-    stype, sversion = read_type(f)
+        check_type_and_version(f, 'TiledCellStMan', 1)
 
-    if stype != 'TiledStMan' or sversion != 2:
-        raise NotImplementedError('Support for {0} version {1} not implemented'.format(stype, sversion))
+        self.default_tile_shape = read_iposition(f)
 
-    st_man = {}
-    st_man['SPEC'] = {}
+        super().read_header(f)
 
-    st_man['BIGENDIAN'] = f.read(1) == b'\x01'  # noqa
 
-    seqnr = read_int32(f)
-    if seqnr != 0:
-        raise ValueError("Expected seqnr to be 0, got {0}".format(seqnr))
-    st_man['SEQNR'] = seqnr
-    st_man['SPEC']['SEQNR'] = seqnr
+class TiledShapeStMan(TiledStMan):
 
-    nrows = read_int32(f)
-    if nrows != 1:
-        raise ValueError("Expected nrows to be 1, got {0}".format(nrows))
+    @classmethod
+    def read(cls, f):
+        self = cls()
+        self.name = 'TiledShapeStMan'
+        return self
 
-    ncols = read_int32(f)
-    if ncols != 1:
-        raise ValueError("Expected ncols to be 1, got {0}".format(ncols))
+    @with_nbytes_prefix
+    def read_header(self, f):
+        check_type_and_version(f, 'TiledShapeStMan', 1)
+        # super().read_header(f)
```

Comment:
am I reading right that this function doesn't do anything except run a check at present?

Reply:
Yes these classes are incomplete, I'll work on them in a separate PR.